control-plane/grpc/proto/v1/pool/pool.proto

```diff
@@ -0,0 +1,151 @@
+syntax = "proto3";
+
+import "v1/misc/common.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+
+package v1.pool;
+
+// A Mayastor Storage Pool
+// It may have a spec which is the specification provided by the creator
+// It may have a state if such state is retrieved from a mayastor storage node
+message Pool {
+  // Desired specification of the pool and metadata
+  optional PoolDefinition definition = 1;
+  // Runtime state of the pool.
+  optional PoolState state = 2;
+}
+
+// Multiple pools
+message Pools {
+  repeated Pool pools = 1;
+}
+
+// Desired spec and the control plane related data
+message PoolDefinition {
+  // Client specification of the pool
+  PoolSpec spec = 1;
+  // Control plane related data for the pool
+  Metadata metadata = 2;
+}
+
+// uuid and the status of pool for the control plane use
```

Comment:
I would suggest rewording `status of pool` to `status of the pool spec` otherwise it may be read as the health of the pool (i.e. online, degraded etc.) rather than whether the pool is being created/deleted. It's the distinction between `SpecStatus` and `PoolStatus`.